gapis/api/vulkan/slice_command_mapper.go

```diff
@@ -22,16 +22,36 @@ import (
 	"github.com/google/gapid/gapis/api/transform"
 )
 
+type beginRenderPassIndex struct {
+	index       uint64
+	renderPass  uint64
+	framebuffer uint64
+}
+
 type sliceCommandMapper struct {
-	subCommandIndicesMap *map[api.CommandSubmissionKey][]uint64
+	subCommandIndicesMap *map[api.CommandSubmissionKey][][]uint64
 	submissionCount      uint64
+	commandBuffersMap    map[uint64]uint64
+	beginRenderPassMap   map[uint64][]beginRenderPassIndex
 }
```

Comment:
I may have missed this on my first view of this CL.

This transform doesn't actually transform anything does it? I can't see where it is actually modifying the commands.
If so, this should probably just be a loop over the commands outside of the transform interface. (That way it wont cost every time we run it, and we can cache the results)